spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigClientProperties.java

```diff
@@ -46,6 +46,16 @@
 	 */
 	public static final String TOKEN_HEADER = "X-Config-Token";
 
+	/**
+	 * Role Id header name.
+	 */
+	public static final String APP_ROLE_ID_HEADER = "X-Config-Role-Id";
```

Comment:
`X-Config-Token` was opaque and can be used for other back ends. Adding vault specific tokens in the client is not a good idea.

Reply:
There is already a `headers` field. Why not make use of that?